cnf-tests/testsuites/e2esuite/sro/sro.go

```diff
@@ -0,0 +1,437 @@
+package sro
+
+import (
+	ctx "context"
+	"fmt"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/execute"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/images"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/nodes"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/pods"
+	srov1beta1 "github.com/openshift-psap/special-resource-operator/api/v1beta1"
+	helmerv1beta1 "github.com/openshift-psap/special-resource-operator/pkg/helmer/api/v1beta1"
+	ocpbuildv1 "github.com/openshift/api/build/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/utils/pointer"
+	"os"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/namespaces"
+	ocpimagev1 "github.com/openshift/api/image/v1"
+)
+
+const (
+	sourceImageStreamName      = "oot-source-driver"
+	sourceImageBuildConfigName = "oot-source-driver-build"
+	ootDriverImageName         = "ice-driver-container"
+	chartName                  = "ice-driver-chart"
+	chartsPath                 = "/usr/src/oot-driver/charts"
+	indexFileName              = "index.yaml"
+	iceDrivTarName             = "ice-driver-0.0.1.tgz"
+	roleName                   = "chart-role"
+	serviceAccountName         = "chart-sa"
+	roleBindingName            = "chart-binding"
+	specialResourceName        = "ice-driver"
+)
+
+var (
+	driverToolKitImage = ""
+)
+
+type buildArgs struct {
+	Name  string
+	Value string
+}
+
+func init() {
+	driverToolKitImage = os.Getenv("DRIVER_TOOLKIT_IMAGE")
+}
+
+var _ = Describe("sro", func() {
+	imageStreamValidation := false
+	configMapValidation := false
+
+	Context("Build source out of tree driver for SRO using", func() {
+		BeforeEach(func() {
```

Comment:
nit (feel free to ignore)
Most of the "BeforeAll/Each" apply to a single item only. Also they usually perform an action, the result of which we check in It(). So maybe they should be part of the It() instead of Before*()?
(feel free to ignore)

Reply:
in general I feel like the `It` should make the verification and all the prework should be done in the BeforeEach (even if there is only 1 validation )